hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -918,7 +918,11 @@ private AbfsRestOperation conditionalCreateOverwriteFile(final String relativePa
       if (e.getStatusCode() == HTTP_CONFLICT) {
         // File pre-exists, fetch eTag
         try {
-          op = client.getPathStatus(relativePath, false, tracingContext);
+          if (getPrefixMode() == PrefixMode.BLOB) {
+            op = client.getBlobProperty(new Path(relativePath), tracingContext);
```

Comment:
Can the path be root here??
We can add a test for that if already not present.

Also, if path is an implicit directory then getBlobProperty will fail with 404 but getPathStatus will not so it can lead to variable behavior...

Reply:
Similar for other places as well...

Reply:
This place can be hit only for file creation.  Have added this handling in the openFileForWrite code for implicit directory handling.